eppo_client/eval.py

```diff
@@ -0,0 +1,92 @@
+from typing import Dict, Optional, Union
+from eppo_client.sharding import Sharder
+from eppo_client.models import Flag, Range, Shard, Variation
+from eppo_client.rules import matches_rule
+from dataclasses import dataclass
+import datetime
+
+
+@dataclass
+class FlagEvaluation:
+    flag_key: str
+    subject_key: str
+    subject_attributes: Dict[str, Union[str, float, int, bool]]
+    allocation_key: str
+    variation: Variation
+    extra_logging: Dict[str, str]
```

Comment:
is `extra_logging` where the evaluation reasoning would work its way in?

Reply:
No, this is a generic object that allows us to add extra logging fields. The immediate use case is to log holdout assignments